internal/isoeditor/rhcos.go

```diff
@@ -326,3 +371,51 @@ func IgnitionImageArchive(ignitionConfig string) ([]byte, error) {
 
 	return compressedBuffer.Bytes(), nil
 }
+
+// Writing the offsets of initrd images in the end of system area (first 32KB).
+// As the ISO template is generated by us, we know that this area should be empty.
+func writeHeader(ignitionOffsetInfo, ramDiskOffsetInfo *OffsetInfo, isoPath string) error {
+	iso, err := os.OpenFile(isoPath, os.O_WRONLY, 0o664)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	// Starting to write from the end of the system area in order to easily support
+	// additional offsets (and as done in coreos-assembler/src/cmd-buildextend-live)
+	headerEndOffset := int64(32768)
+
+	// Write ignition config
+	headerEndOffset, err = writeOffsetInfo(headerEndOffset, ignitionOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	// Write ram disk
+	_, err = writeOffsetInfo(headerEndOffset, ramDiskOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeOffsetInfo(headerOffset int64, offsetInfo *OffsetInfo, iso *os.File) (int64, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, binary.LittleEndian, offsetInfo)
+	if err != nil {
+		return 0, err
+	}
+
+	headerOffset = headerOffset - int64(buf.Len())
+	_, err = iso.Seek(headerOffset, io.SeekStart)
+	if err != nil {
+		return 0, err
+	}
+	_, err = iso.Write(buf.Bytes())
+	if err != nil {
+		return 0, err
+	}
+
+	return headerOffset, nil
```

Comment:
This is strange to me ... I feel like a more "standard" way to do this would be to return the bytes written then the caller can determine the new starting point.

Reply:
ok, so i'll just return the length of written bytes to make it clearer.